src/cargo/core/features.rs

```diff
@@ -653,7 +653,7 @@ unstable_cli_options!(
     no_index_update: bool = ("Do not update the registry index even if the cache is outdated"),
     panic_abort_tests: bool = ("Enable support to run tests with -Cpanic=abort"),
     host_config: bool = ("Enable the [host] section in the .cargo/config.toml file"),
-    http_registry: bool = ("Support HTTP-based crate registries"),
+    sparse_registry: bool = ("Support plain-HTTP-based crate registries"),
```

Comment:
I've deliberately used a different phrase here to teach users what sparse means, otherwise the help would be a variant of "--enable-flubbubl: Enables flubbubl".